packages/studiocms_ui/src/integration.ts

```diff
@@ -14,17 +23,70 @@ export default function integration(options: Options = {}): AstroIntegration {
 		`import '${resolve('./css/global.css')}'`
 	);
 
+	const version = viteVirtualModulePluginBuilder(
+		'studiocms:ui/version',
+		'sui-version',
+		`export default '${packageVersion}'`
+	);
+	
 	return {
 		name: '@studiocms/ui',
 		hooks: {
-			'astro:config:setup': ({ injectScript, updateConfig }) => {
+			'astro:config:setup': (params) => {
+				const { addDevToolbarApp, injectScript, updateConfig } = params;
+				const { resolve: rootResolve } = createResolver(params.config.root.pathname);
+
 				updateConfig({
 					vite: {
-						plugins: [globalCss()],
+						plugins: [globalCss(), version()],
+					},
+				});
+
+				addVirtualImports(params, {
+					name: 'sui-scripts',
+					imports: {
+						'studiocms:ui/scripts/checkbox': `import ${JSON.stringify(resolve('./components/Checkbox/checkbox.ts'))}`,
+						'studiocms:ui/scripts/radiogroup': `import ${JSON.stringify(resolve('./components/RadioGroup/radiogroup.ts'))}`,
+						'studiocms:ui/scripts/searchselect': `import ${JSON.stringify(resolve('./components/SearchSelect/searchselect.ts'))}`,
+						'studiocms:ui/scripts/select': `import ${JSON.stringify(resolve('./components/Select/select.ts'))}`,
+						'studiocms:ui/scripts/tabs': `import ${JSON.stringify(resolve('./components/Tabs/tabs.ts'))}`,
+						'studiocms:ui/scripts/themetoggle': `import ${JSON.stringify(resolve('./components/ThemeToggle/themetoggle.ts'))}`,
+						'studiocms:ui/scripts/toaster': `import ${JSON.stringify(resolve('./components/Toast/toaster.ts'))}`,
+						'studiocms:ui/scripts/toast': `import ${JSON.stringify(resolve('./components/Toast/toast.ts'))}`,
+						'studiocms:ui/scripts/toggle': `import ${JSON.stringify(resolve('./components/Toggle/toggle.ts'))}`,
+						'studiocms:ui/scripts/accordion': `import ${JSON.stringify(resolve('./components/Accordion/accordion.ts'))}`,
 					},
 				});
 
 				injectScript('page-ssr', `import 'studiocms:ui/global-css';`);
+
+				if (options.customCss) {
+					const customCss = viteVirtualModulePluginBuilder(
+						'studiocms:ui/custom-css',
+						'sui-custom-css',
+						`import '${rootResolve(options.customCss)}'`
+					);
+
+					updateConfig({
+						vite: {
+							plugins: [customCss()]
+						}
+					});
+
+					injectScript('page-ssr', `import 'studiocms:ui/custom-css';`);
+				}
+
+				addDevToolbarApp({
+					id: 'studiocms:ui/toolbar',
+					name: 'StudioCMS/UI',
+					entrypoint: '@studiocms/ui/toolbar',
```

Comment:
```suggestion
					entrypoint: resolve('./toolbar/index.ts'),
```
Should be using resolver when possible to prevent a issue when the UI lib is used as a dep of a dep :sweat_smile: